### PR TITLE
[bot] name -> pname (easy cases)

### PR DESCRIPTION
**Pull request by volth**
treewide replacement of 
```nix
stdenv.mkDerivation rec {
  name = "*-${version}";
  version = "*";
```
to `pname`

cc @worldofpeace #61691

### CODE REVIEW

**Comment by mmahut** [comment]
Is this worth the huge rebuild? We should probably be fixing it over time with updates?

**Comment by worldofpeace** [comment]
@mmahut I think we should start to getting everything using pname if we can.

**Comment by mmahut** [comment]
> @mmahut I think we should start to getting everything using pname if we can.

I agree. Maybe we can create an ``GrahamcOfBorg`` that will check if new merges are fixing the pname as well? Or maybe just one large rebuild will do? In that case, maybe it should go to ``staging`` first?

**Comment by worldofpeace** [comment]
@mmahut I'm guessing that's why volth closed this.

>  Or maybe just one large rebuild will do?

Since we group mass rebuilds together into `staging-next` we're going to do a mass rebuild continuously anyway.

File: pkgs/applications/altcoins/nano-wallet/default.nix [hunk 1]
```diff
@@ -29,7 +29,7 @@ stdenv.mkDerivation rec {
       Qt5Gui_INCLUDE_DIRS = "${qtbase.dev}/include/QtGui";
       Qt5Widgets_INCLUDE_DIRS = "${qtbase.dev}/include/QtWidgets";
     };
-    optionToFlag = name: value: "-D${name}=${value}";
+    optionToFlag = name: value: "-D${pname}-${version}=${value}";
```
Comment (jtojnar) [inline pkgs/applications/altcoins/nano-wallet/default.nix]: Misreplacement.

Comment (volth) [reply pkgs/applications/altcoins/nano-wallet/default.nix]: thanks!

**Comment by volth** [comment]
The script (I used one from https://github.com/NixOS/nixpkgs/pull/63177) undestands lexical scopes, but I did not expect so many wrong replacemens because of name `name` used in nested scopes.
So I had to imlement it anyway, rerun the script and add the errata here.

**Comment by volth** [comment]
oh, too late.
merge conflict

**Comment by volth** [comment]
I do not know what to do to avoid merge conflicts while a big PR is being reviewed.
Slice it so smaller PR's, say with 100 files each?
That will be 47 PRs only in this bot pass, which is about 40% of required replacements, so 100+ PRs at the end.

**Comment by volth** [comment]
@Mic92 maybe also https://github.com/volth/nixpkgs-windows/commit/e82ae097c424fbbd92f0c4176a66542d08fe1f09

**Comment by Mic92** [comment]
@volth can you open this in a PR for the sake of eval checks?

**Comment by volth** [comment]
> @volth can you open this in a PR for the sake of eval checks?

https://github.com/NixOS/nixpkgs/pull/66751

**Pull request merged**